### ISSUE TITLE
Different validation results EVS Client vs matchbox GUI

### ISSUE THREAD

**Issue by oliveregger** [ahdis/matchbox#270]
https://ehealthsuisse.ihe-europe.net/evs/report.seam?oid=1.3.6.1.4.1.12559.11.25.4.844009

we get warnings for message:

Constraint ID
INVALID
Constraint Type
TerminologyEngine
Constraint Description
Wrong Display Name 'Laboratory report' for http://loinc.org#11502-2. Valid display is one of 0 choices: 'Laboratory report' (en-US), 'Laboratory report' (en-US), 'Laboratory report' (en-US), 'Y' (en-US), 'Reporte de laboratorio: Paciente:Punto temporal:Tipo:Documento:' (es-MX), 'informe del laboratorio.total:hallazgo:duración del encuentro:^paciente:Múltiple:' (es-AR), 'Laborbericht' (de-DE), 'Laboratoriumrapport [bevinding] (document)' (nl-NL) or 'Laborbefund' (de-AT) (for the language(s) 'fr-CH')
Location in validated object
line 47, column 14, FHIRPath: DocumentReference.contained/*Bundle/8ee5d863-d208-46ed-a638-ff605a41e89b*/.entry[0].resource.type.coding[1].display

the sentence looks strange: Valid display is one of 0 choices, but there are multiple choices listed (and we have the value inside).

it cannot be reproduced directly in matchbox with the GUI, we don't get those warnings.

**Comment by qligier** [comment]
In the tx server calls:

Request
```json
{
  "resourceType": "Parameters",
  "parameter": [
    {
      "name": "coding",
      "valueCoding": {
        "system": "http://loinc.org",
        "code": "11502-2",
        "display": "Laboratory report"
      }
    },
    {
      "name": "displayLanguage",
      "valueString": "english"
    },
    {
      "name": "lenient-display-validation",
      "valueBoolean": true
    },
    {
      "name": "default-to-latest-version",
      "valueBoolean": true
    },
    {
      "name": "cache-id",
      "valueId": "012a7f74-258d-433e-aa3a-c3ff26dd1e1e"
    },
    {
      "name": "profile-url",
      "valueString": "http://hl7.org/fhir/ExpansionProfile/dc8fd4bc-091a-424a-8a3b-6198ef146891"
    },
    {
      "name": "mode",
      "valueString": "lenient-display-validation"
    }
  ]
}
```

Response:
```json
{
  "resourceType": "Parameters",
  "parameter": [
    {
      "name": "result",
      "valueBoolean": true
    },
    {
      "name": "system",
      "valueUri": "http://loinc.org"
    },
    {
      "name": "code",
      "valueCode": "11502-2"
    },
    {
      "name": "version",
      "valueString": "2.78"
    },
    {
      "name": "display",
      "valueString": "Laboratory report"
    },
    {
      "name": "message",
      "valueString": "Wrong Display Name 'Laboratory report' for http://loinc.org#11502-2. Valid display is one of 0 choices: 'Laboratory report' (en-US), 'Laboratory report' (en-US), 'Laboratory report' (en-US), 'Y' (en-US), 'Reporte de laboratorio: Paciente:Punto temporal:Tipo:Documento:' (es-MX), 'informe del laboratorio.total:hallazgo:duración del encuentro:^paciente:Múltiple:' (es-AR), 'Laborbericht' (de-DE), 'Laboratoriumrapport [bevinding] (document)' (nl-NL) or 'Laborbefund' (de-AT) (for the language(s) 'english')"
    },
    {
      "name": "issues",
      "resource": {
        "resourceType": "OperationOutcome",
        "issue": [
          {
            "severity": "warning",
            "code": "invalid",
            "details": {
              "coding": [
                {
                  "system": "http://hl7.org/fhir/tools/CodeSystem/tx-issue-type",
                  "code": "invalid-display"
                }
              ],
              "text": "Wrong Display Name 'Laboratory report' for http://loinc.org#11502-2. Valid display is one of 0 choices: 'Laboratory report' (en-US), 'Laboratory report' (en-US), 'Laboratory report' (en-US), 'Y' (en-US), 'Reporte de laboratorio: Paciente:Punto temporal:Tipo:Documento:' (es-MX), 'informe del laboratorio.total:hallazgo:duración del encuentro:^paciente:Múltiple:' (es-AR), 'Laborbericht' (de-DE), 'Laboratoriumrapport [bevinding] (document)' (nl-NL) or 'Laborbefund' (de-AT) (for the language(s) 'english')"
            },
            "location": [
              "Coding.display"
            ],
            "expression": [
              "Coding.display"
            ]
          }
        ]
      }
    }
  ]
}
```

**Comment by oliveregger** [comment]
- matchboxv3 on test.ahdis.ch is using internal terminology server thats we don't see the error there
- tx.fhir.org is enforcing the validation of the display name according to the language
- for the first erros it is using composition.language and then the display name is not correct
- for the other erros the validaor is falling back to english (from the context), this should be en-US and not english

**Comment by qligier** [comment]
'english' fixed in b5c05e1d12c89b72aa7f00bda4a76523b3530501

**Comment by oliveregger** [comment]
Need to add a testcase in matchbox-server that the gazelle API call is running.

**Comment by oliveregger** [comment]
fixed with 3.9.3
